pkg/kwokctl/runtime/util_test.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package runtime
+
+import (
+	"reflect"
+	"testing"
+
+	"sigs.k8s.io/kwok/pkg/apis/internalversion"
+)
+
+func TestApplyComponentArgsOverride(t *testing.T) {
+	tests := []struct {
+		name     string
+		patch    internalversion.ExtraArgs
+		wantArgs []string
+	}{
+		{
+			patch: internalversion.ExtraArgs{
+				Key:      "etcd-servers",
+				Value:    "http://192.168.66.2:2379",
+				Override: true,
+			},
+			name:     "Override the value",
+			wantArgs: []string{"--etcd-servers=http://192.168.66.2:2379", "--etcd-prefix=/registry"},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			c := newDefault()
+			applyComponentArgsOverride(c.Args, internalversion.ExtraArgs{
+				Key:      tt.patch.Key,
+				Value:    tt.patch.Value,
+				Override: tt.patch.Override,
+			})
+
+			if !reflect.DeepEqual(tt.wantArgs, c.Args) {
+				t.Errorf("Exist is not expact! key:%s want args:%s, got args:%s", tt.patch.Key, tt.wantArgs, c.Args)
+			}
+		})
+	}
+}
+
+func TestApplyComponentPatch(t *testing.T) {
+	tests := []struct {
+		name     string
+		patch    internalversion.ExtraArgs
```

Comment:
```suggestion
		patch    internalversion.ComponentPatches
```

Here should be test with `internalversion.Component` and`internalversion.ComponentPatches`